eZ/Publish/Core/Base/Tests/Container/Compiler/TaggedServiceIdsIterator/DeprecationErrorCaptor.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\Core\Base\Tests\Container\Compiler\TaggedServiceIdsIterator;
+
+/**
+ * Captures user deprecation warnings emitted using trigger_error function.
+ *
+ * @internal
+ */
+final class DeprecationErrorCaptor
```

Comment:
Does this captor demand a ransom? ;)

I'm not really sold on this naming, the most commonly used name for this kind of classes is Collector, I think. Unless this follows some kind of existing pattern?

General question: why do you need to add extra complexity to error handling? We rather rely on Monolog and Symfony Console handlers on that. Is it just to be able to process those errors in unit tests? For that we usually rely on warnings being converted to exceptions and validate them by `->expectException`. Doesn't this work here?

Reply:
> I'm not really sold on this naming, the most commonly used name for this kind of classes is Collector, I think. Unless this follows some kind of existing pattern?

It was inspirated by https://www.javadoc.io/doc/org.mockito/mockito-core/latest/org/mockito/ArgumentCaptor.html but collector seems to be better naming choice,

> General question: why do you need to add extra complexity to error handling? We rather rely on Monolog and Symfony Console handlers on that. Is it just to be able to process those errors in unit tests? For that we usually rely on warnings being converted to exceptions and validate them by ->expectException. Doesn't this work here?

There are dedicated `->expectDeprecation(...)`, `->expectDeprecationMessage(...)` but for some reason they doesn't work. I need to figure out why.

Reply:
`->expectDeprecation(...)` and `->expectDeprecationMessage(...)` only works when the deprecation message is triggered without error suppression operator (\`@\`) e.g.  
 

```php
trigger_error('Lorem ipsum dolor...', E_USER_DEPRECATED)   // OK
@trigger_error('Lorem ipsum dolor...', E_USER_DEPRECATED)  // KO
```